script.rb

```diff
@@ -0,0 +1,155 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    while i < length
+      temp << self[i] if yield(self[i])
+      i += 1
+    end
+    temp
+  end
+
+  def my_all?
+    i = 0
+    while i < length
+      return true if yield(self[i]) && yield(self[i + 1])
+
+      i += 1
+      return false
+    end
+  end
+
+  def my_any?
+    i = 0
+    while i < length
+      return true if yield(self[i]) || yield(self[i + 1])
+
+      i += 1
+      return false
+    end
+  end
+
+  def my_none?
+    i = 0
+    while i < length
+      return true unless yield(self[i])
+
+      i += 1
+      return false
+    end
+  end
+
+  def my_count
+    cont = 0
+    i = 0
+    while i < length
+      cont += 1 if yield(self[i])
+      i += 1
+    end
+    cont
+  end
+
+  def my_map(my_proc = nil)
+    temp = []
+    i = 0
+    while i < length
+      temp << if my_proc
+                my_proc.call(self[i])
+              else
+                yield(self[i])
+              end
+      i += 1
+    end
+    temp
+  end
```

Comment:
- For my_map returns an Enumerator if no block is given.
 Built-in method
`[1,2,3].map` returns `=> #<Enumerator: [1, 2, 3]:map>
`

Your method gives the following error

> LocalJumpError:
        no block given (yield)

Reply:
my_map method updated